### PR TITLE
Adds vertical variation to navigation menu

### LINKED ISSUE #16829: Navigation Block: add support for vertical menus
Both horizontal (e.g. in a header) and vertical (e.g. in a sidebar) menus should be supported by the Navigation block.

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Fixes #16829

## How has this been tested?
Tested locally by:

1. Add a post
2. Use the top inserter button
3. Search for vertical
4. Insert the block Verical Navigation

## Screenshots

vertical-menu

## Types of changes
Non breaking change that introduces a vertical variation to the Navigation block.

### CODE REVIEW

**Review by talldan — COMMENTED** [review]
This looks pretty good, and it's great that it's only a small change.

There were a couple of things that I noticed, which are mostly comments on how variations work in general.

I think this needs a good copy/design review. I wonder if 'Navigation (Horizontal)' and 'Navigation (Vertical)' might be better titles in the context of the inserter:
Screenshot 2020-04-01 at 11 55 45 am

The second thing is that it feels like there should be a transform or a way to switch between variation. It looks like that's tracked by https://github.com/WordPress/gutenberg/issues/20584, and it should probably happen automatically through declaration of the variation.

Also, that the variations don't appear in the slash inserter confused me, but looks like that's tracked by https://github.com/WordPress/gutenberg/issues/20583.

**Comment by michelleweber** [comment]
I would +1 'Navigation (Horizontal)' and 'Navigation (Vertical).'

**Review by talldan — APPROVED** [review]
Approving from a code point of view.

There might still be some unanswered design questions, and I think the other question relevant for variations is how can they be exposed for an existing block that wants to switch from horizontal to vertical.

**Comment by aristath** [comment]
Shouldn't this be just an option inside a single navigation block? All that's really needed is to add a class to the element to flag if it's horizontal or vertical, so a simple radio-input in the navigation block to select orientation would do just the same

**Comment by draganescu** [comment]
hi @aristath having this as an attribute works better as this setting influences other behaviors as well (for example mover direction). The way to have attributes preset is to use the new variations API. There was some discussion about using block styles but it would complicate styling for every single difference that the vertical or horizontal layout makes.

**Comment by mtias** [comment]
This is cool, a nice small tweak. cc @gziolo for the variations nod. Next step is to allow transforms!

File: packages/block-library/src/navigation/index.js [hunk 1]
```diff
@@ -32,6 +32,22 @@ export const settings = {
 		lightBlockWrapper: true,
 	},
 
+	variations: [
```
Comment (gziolo) [inline packages/block-library/src/navigation/index.js:35]: Nice one, it's so useful 👍

**Comment by gziolo** [comment]
> cc @gziolo for the variations nod. Next step is to allow transforms!

Do you think, we should allow transforms for variations?

**Comment by mtias** [comment]
For some, absolutely. It's a way to "swap" one variation with another. It could be useful for changing a Twitter profile with Mastodon, or to change the layout arrangement of columns. It might also be exposed in its own "Variations" panel inside the transforms menu (like styles are).

**Comment by gziolo** [comment]
> For some, absolutely. It's a way to "swap" one variation with another. It could be useful for changing a Twitter profile with Mastodon, or to change the layout arrangement of columns. It might also be exposed in its own "Variations" panel inside the transforms menu (like styles are).

#Yes, I agree on that changes on attributes would be quite useful and in general safe to apply. With Columns block and inner blocks case we would need to decide if we provide integration with callback functions that help avoiding content loss. The implementation itself doesn’t look complex unless we want to keep the current variation highlighted in UI.

**Comment by ellatrix** [comment]
This seems to have caused a regression in master. The movers for the normal navigation block are no longer horizontal.

**Comment by strarsis** [comment]
I just wanted to create a new issue for this, glad I noticed.
When will you publish a new Gutenberg release?

Related issue: https://github.com/WordPress/gutenberg/issues/22085

**Pull request merged**